secondLayout.js

```diff
@@ -0,0 +1,317 @@
+// Getting the canvas from html page
+let canvas = document.getElementById("canvas");
+let ctx = document.getElementById("canvas").getContext("2d");
+let bodyCtx = document.querySelector("body");
+
+window.onload = () => {
+  firstPage();
+};
+
+function startGame() {
+  isOutOfFirstPage = true;
+  drawGameBoard();
+}
+
+// to draw the first page with bg and text on canvas
+function firstPage() {
+  drawFirstBg();
+  let newText = new text(gameStory, 100, 150, 35);
+  newText.update(100);
+}
+
+// function to remove objects from the objects array and from the canvas when they reach the end of the canvas
+function removeObject(objectsArr) {
+  for (let i = 0; i < objectsArr.length; i++) {
+    if (objectsArr[i].y >= 700) {
+      let index = objectsArr.indexOf(objectsArr[i]);
+      objectsArr.splice(index, 1);
+    }
+    // remove the obstacles
+    if (
+      objectsArr[i] != undefined &&
+      objectsArr[i].scoreType == "obstacles" &&
+      objectsArr[i].x >= 1200
+    ) {
+      let index1 = objectsArr.indexOf(objectsArr[i]);
+      objectsArr.splice(index1, 1);
+    }
+  }
+}
+
+// function to check if the new generated object collied with any of the items in the objects arry  to avoid
+// objects overlapping
+function crashesWithAnything(obj) {
+  for (i = 0; i < objectsArr.length; i++) {
+    if (obj.crash(objectsArr[i])) {
+      return true;
+    }
+  }
+  return false;
+}
+
+// draw function for the page after winning or losing the game
+let drawEndGame = () => {
+  ctx.clearRect(0, 0, canvas.width, canvas.height);
+  // draw the bg
+  // draw img if the player won
+  if (!runningGame && !winGame) {
+    drawEndGameBg();
+    let winImg = new Image();
+    winImg.src = "images/You-win.png";
+    ctx.drawImage(winImg, 400, 300, 400, 50);
+    // draw another img if the player lost
+  } else {
+    bodyCtx.style.backgroundImage =
+      "url('images/stars.png'),url('images/endGameBg/background.png')";
+    drawEndGameBg();
+    let winImg = new Image();
+    winImg.src = "images/lose.png";
+    ctx.drawImage(winImg, 400, 300, 400, 50);
+  }
+  // press space to play the game again
+
+  window.requestAnimationFrame(drawEndGame);
```

Comment:
I guess you don't have to continuously re-draw the canvas after the game has ended (unless something is moving on that end-game-canvas)